### PR TITLE
Update gitignore

### PR DESCRIPTION
**Pull request by SpecialYang**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -54,3 +54,5 @@ etc/
 var/
 # Go compiled tests
 *.test
+# OSX trash
+.DS_Store
```
Comment (ericvn) [inline .gitignore:58]: You probably want to ignore the file in all directories: **/.DS_Store

Comment (ericvn) [inline .gitignore:58]: I'm not sure what is causing these files to be created, so not sure if the other extraneous MacOS files should also be included.

Comment (SpecialYang) [reply .gitignore:58]: According to this [site](https://buildthis.com/ds_store-files-and-why-you-should-know-about-them/), we should avoid adding .DS_Store files into repo. In other well-known git repositories, such as kubernetes and helm, .DS_Store files are added to the `.gitignore` file.

Comment (ericvn) [reply .gitignore:58]: If we want to follow Kubernetes, we should add the other MacOS (I would update the comment to the rename MacOS) entry as well:
```
# MacOS leaves these everywhere on SMB shares
._*

# MacOS Desktop Services Store
.DS_Store
```

Comment (SpecialYang) [reply .gitignore:60]: LGTM. Thanks.

**Review by ericvn — APPROVED** [review]

**Pull request merged**